lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,43 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    key = get_correct_key(method_name)
+    lazy?(key) ? self[key] = self[key].call : self[key]
+  end
+
+  def lazy?(key)
+    key = check_correct_key(key)
```

Comment:
Good solution. But `check_correct_key` could return `nil` if there is no required key in a hash.
So, you can add `return false unless key`line to not receive value of an absent key.